.github/workflows/generate.yaml

```diff
@@ -0,0 +1,52 @@
+name: Generate and Test on OAS change
+
+on:
+  pull_request:
+    paths:
+      - 'openapi.json'
+      - 'api.yaml'
+    types:
+      - opened
+
+jobs:
+  generate_and_test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      
+    - name: Set up Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - name: Install dependencies
+      run: npm install -g konfig-cli
+
+    - name: Regenerate SDKs
+      run: konfig generate
+
+    - name: Create changeset file
+      run: konfig changeset -a -m "Regenerate SDKs"
+
+    - name: Commit changes
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git commit -am "Regenerate SDKs"
+      
+    - name: Run tests
+      run: konfig test
+
+    - name: Get PR Number
+      id: get_pr_number
+      run: |
+        PR_JSON=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/pulls?head=${{ github.ref }}")
+        echo "::set-output name=PR_NUMBER::$(echo "$PR_JSON" | python3 -c "import sys, json; print(json.load(sys.stdin)[0]['number'])")"
+
+    - name: Merge PR if tests pass
+      if: success()
+      run: |
+        PR_NUMBER=${{ steps.get_pr_number.outputs.PR_NUMBER }}
+        curl -X PUT -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/pulls/$PR_NUMBER/merge"
```

Comment:
chat GPT wrote this
I think other alternative would be using an action from the marketplace to merge PR instead of using curl

Reply:
can we do this inside a CLI command instead? `konfig pull-request`

Reply:
that way can u can test this his locally with a [PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)

Reply:
sure thing.
By "this", do you mean just these lines highlighted in the comment? or did you want the entire github action essentially in a single CLI command

Reply:
its nice to split up steps in the github action so you can see which step failed so I was just gonna say "this" is the highlighted parts.

But I realize PR_NUMBER is specifically a variable inside the PR so it would be hard to test locally, right? so it would be hard to test

Reply:
yeah - we could test using acme maybe
but $PR_NUMBER is just an environment variable right? So it could still be referenced in the CLI command, even if it isn't set locally? is that true?

Reply:
Actually yeah and the PR number is easy to get

Reply:
ok, I'll move forward with the CLI command for these steps then. Minor - What about `konfig merge-pull-request`? To me `konfig pull-request` implies opening a PR.

Reply:
yeah merge in the name makes sense